app/data_client.go

```diff
@@ -0,0 +1,781 @@
+func (d *Client) DeleteBinaryDataByFilter(ctx context.Context, filter Filter) (uint64, error) {
+	resp, err := d.client.DeleteBinaryDataByFilter(ctx, &pb.DeleteBinaryDataByFilterRequest{
+		Filter:              filterToProto(filter),
+		IncludeInternalData: true,
+	})
+	if err != nil {
+		return 0, err
+	}
+	return resp.DeletedCount, nil
+}
+
+// DeleteBinaryDataByIDs deletes binary data based on given IDs.
+func (d *Client) DeleteBinaryDataByIDs(ctx context.Context, binaryIDs []BinaryID) (uint64, error) {
+	resp, err := d.client.DeleteBinaryDataByIDs(ctx, &pb.DeleteBinaryDataByIDsRequest{
+		BinaryIds: binaryIDsToProto(binaryIDs),
+	})
+	if err != nil {
+		return 0, err
+	}
+	return resp.DeletedCount, nil
+}
+
+// AddTagsToBinaryDataByIDs adds string tags, unless the tags are already present, to binary data based on given IDs.
+func (d *Client) AddTagsToBinaryDataByIDs(ctx context.Context, tags []string, binaryIDs []BinaryID) error {
+	_, err := d.client.AddTagsToBinaryDataByIDs(ctx, &pb.AddTagsToBinaryDataByIDsRequest{
+		BinaryIds: binaryIDsToProto(binaryIDs),
+		Tags:      tags,
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// AddTagsToBinaryDataByFilter adds string tags, unless the tags are already present, to binary data based on the given filter.
+func (d *Client) AddTagsToBinaryDataByFilter(ctx context.Context, tags []string, filter Filter) error {
+	_, err := d.client.AddTagsToBinaryDataByFilter(ctx, &pb.AddTagsToBinaryDataByFilterRequest{
+		Filter: filterToProto(filter),
+		Tags:   tags,
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// RemoveTagsFromBinaryDataByIDs removes string tags from binary data based on given IDs.
+func (d *Client) RemoveTagsFromBinaryDataByIDs(ctx context.Context,
+	tags []string, binaryIDs []BinaryID,
+) (uint64, error) {
+	resp, err := d.client.RemoveTagsFromBinaryDataByIDs(ctx, &pb.RemoveTagsFromBinaryDataByIDsRequest{
+		BinaryIds: binaryIDsToProto(binaryIDs),
+		Tags:      tags,
+	})
+	if err != nil {
+		return 0, err
+	}
+	return resp.DeletedCount, nil
+}
+
+// RemoveTagsFromBinaryDataByFilter removes string tags from binary data based on the given filter.
+func (d *Client) RemoveTagsFromBinaryDataByFilter(ctx context.Context,
+	tags []string, filter Filter,
+) (uint64, error) {
+	resp, err := d.client.RemoveTagsFromBinaryDataByFilter(ctx, &pb.RemoveTagsFromBinaryDataByFilterRequest{
+		Filter: filterToProto(filter),
+		Tags:   tags,
+	})
+	if err != nil {
+		return 0, err
+	}
+	return resp.DeletedCount, nil
+}
+
+// TagsByFilter gets all unique tags from data based on given filter.
+func (d *Client) TagsByFilter(ctx context.Context, filter Filter) ([]string, error) {
+	resp, err := d.client.TagsByFilter(ctx, &pb.TagsByFilterRequest{
+		Filter: filterToProto(filter),
+	})
+	if err != nil {
+		return nil, err
+	}
+	return resp.Tags, nil
+}
+
+// AddBoundingBoxToImageByID adds a bounding box to an image with the given ID.
+func (d *Client) AddBoundingBoxToImageByID(
```

Comment:
I think we need a bit more clarification in the docstring on how this works. the float values, e.g., _must_ be somewhere between 0 and 1, but that's not obvious based on what we have here. I don't think we need as much detail as exists in the python SDK, but taking a peek there might be useful for more context to provide.